### PR TITLE
[Feature] Skills quick summary sidebar

### LINKED ISSUE #8213: ✨ Skills quick summary sidebar
## ✨ Feature

The assessment plan builder page should have a quick summary sidebar.

Quick notes:
- Mostly behave the same as the table of contents
- They are a mini summary of what's on the table
- If a skill is missing an assessment then it is red
- If it has at least one assessment then it's green
- Warning on the total - warning if too few assessments (1) or too many (4+)
- The link at the bottom scrolls to the skill summary section on that page

## 🎨 Design File and Designer

[Link to figma file](https://www.figma.com/file/em5ArPr044YaIe1Vb1zCZz/Pools---Record-of-Decision?type=design&node-id=5437%3A34317&mode=design&t=11Dob6JX6BOb3lle-1) by @Jerryescandon 

## 📸 Screenshot

image

## ✅ Acceptance Criteria

- [ ] Sidebar added to page
- [ ] Received data and correctly displays output

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #8213 

## 👋 Introduction

This branch adds the Skills quick summary sidebar to the work-in-progress Assessment plan builder page.  

## 🕵️ Details

There's a bit of extra refactoring in this branch to make it work:
- I've factored out the Sidebar component from the TableOfContents component.
- I've added a _warning_ status and a compact layout to the StatusItem component.

## 🧪 Testing

- [ ] Existing uses of StatusItem and TableOfContents look OK
- [ ] Sidebar on `/en/admin/pools/:poolId/plan` matches design
- [ ] Validation on status items works
  - [ ] Total goes to warning if not 1-4
  - [ ] Skills go to error if 0

Here are some GraphQL queries to help with validation testing.

```graphql
query PoolQuery {
  pool(id: "1adf903a-ff4b-4018-9e2e-cbf33a5823de") {
    assessmentSteps {
      id
      title {
        en
      }
      poolSkills {
        id
        skill {
          name {
            en
          }
        }
      }
    }
    poolSkills {
      id
      skill {
        name {
          en
        }
      }
    }
  }
}
```

```graphql
mutation UpdateAssessmentStep($id: UUID!, $step: AssessmentStepInput) {
  updateAssessmentStep(id: $id, assessmentStep: $step) {
    id
  }
}
```

## 📸 Screenshot

image

### CODE REVIEW

File: apps/web/src/components/StatusItem/StatusItem.tsx [hunk 1]
```diff
@@ -106,10 +126,8 @@ const StatusItem = ({
   return (
     <Wrapper
       data-h2-display="base(flex)"
-      data-h2-border-top="base:selectors[:not(:first-child)](1px solid gray.lighter)"
-      data-h2-margin-top="base:selectors[:not(:first-child)](x.5)"
-      data-h2-padding-top="base:selectors[:not(:first-child)](x.5)"
       data-h2-justify-content="base(space-between)"
+      {...layoutStyleMap[layout]}
```
Comment (esizer) [inline apps/web/src/components/StatusItem/StatusItem.tsx:130]: Can we add a tiny `gap` on this to space the main text out a bit from the item count?

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx [hunk 2]
```diff
@@ -16,6 +16,8 @@ import {
   Pending,
   Separator,
 } from "@gc-digital-talent/ui";
+import Sidebar from "@gc-digital-talent/ui/src/components/Sidebar";
```
Comment (esizer) [inline apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx:19]: If you add `Sidebar` to the `ui` barrel export file we can add this to the import statement above.

**Review by esizer — CHANGES_REQUESTED** [review]
Looks great! Two _very_ minor suggestions 👍

File: apps/web/src/components/StatusItem/StatusItem.tsx (hunk 1 above)
Comment (petertgiles) [reply apps/web/src/components/StatusItem/StatusItem.tsx]: Good idea, yeah.

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx (hunk 2 above)
Comment (petertgiles) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/AssessmentPlanBuilderPage.tsx]: Oh yeah, I like that.
[add sidebar to barrel export](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8290/commits/98d648d1ef096773507435cd63f739fc11906bbe)

**Review by esizer — APPROVED** [review]
Cool! 😎

**Pull request merged**